data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/MutateMapper.java

```diff
@@ -54,11 +55,13 @@ public RenameCopyConfig(String fromKey, String toKey) {
                     renames.add(new RenameCopyConfig(entry.getKey(), entry.getValue()));
                 });
             } else if(Objects.equals(name, "remove_field")) {
-                deletes.add((ArrayList<String>)attr.getAttributeValue().getValue());
+                deletes.addAll((ArrayList<String>)attr.getAttributeValue().getValue());
             } else if(Objects.equals(name, "copy")) {
                 ((Map<String, String>)attr.getAttributeValue().getValue()).entrySet().forEach(entry -> {
                     copies.add(new RenameCopyConfig(entry.getKey(), entry.getValue()));
                 });
+            } else if(Objects.equals(name, "uppercase")) {
+                uppercases.addAll((ArrayList<String>)attr.getAttributeValue().getValue());
```

Comment:
This class is doing multiple things and not following single responsibility principle. Consider creating interface and implementation for each string processor in separate class to make this class more maintainable.

You can have use DI to inject Map of StringProcessorType -> Implementation. This class can lookup the Map for StringProcessor type and call the interface method. This will keep the code more logical and maintainable.

Reply:
Agreed. I mentioned changing this class to make it more like how you are saying during standup, but I want to get a sample processor and the base structure out for @sshivanii to build off of first.